controllers/managedclustermodule_reconciler.go

```diff
@@ -82,25 +87,35 @@ func (r *ManagedClusterModuleReconciler) Reconcile(ctx context.Context, req ctrl
 
 	logger.Info("Requested KMMO ManagedClusterModule")
 
-	clusters, err := r.getSelectedManagedClusters(ctx, mcm)
+	clusters, err := r.clusterAPI.SelectedManagedClusters(ctx, mcm)
 	if err != nil {
 		return res, fmt.Errorf("failed to get selected clusters: %v", err)
 	}
 
 	for _, cluster := range clusters.Items {
+		logger := log.FromContext(ctx).WithValues("cluster", cluster.Name)
+		clusterCtx := log.IntoContext(ctx, logger)
+
+		if requeue := r.clusterAPI.BuildAndSign(clusterCtx, *mcm, cluster); requeue {
+			logger.Info("Build and Sign require a requeue; skipping ManifestWork reconciliation")
+			res.Requeue = true
+			continue
```

Comment:
Not sure about the inner implementation but does it mean that we are not going to update the status of other clusters as long as a prior one has been requed?

Reply:
The goal of enabling the `res.Requeue` and then `continue`-ing through the loop is to indicate that this managed cluster cannot move forward with the deployment of the KMM `Module` via the creation/patching of the respective OCM `ManifestWork` CR. One or more `{build,sign}` Jobs for this `Module` might be still running. So we just indicate that we need to have the specified `ManagedClusterModule` CR go into a additional reconcile loop, even if only one of the managed clusters needs to wait for a `{build/sign}` Job to complete.

WDYT?

Reply:
Yup. Sound good to me. Thanks for the explanation.